edgelet/aziot-edged/src/product_info.rs

```diff
@@ -0,0 +1,194 @@
+use std::convert::AsRef;
+use std::fmt;
+use std::fs;
+use std::io::{self, BufRead};
+use std::path::Path;
+use std::process::{Command, Output};
+
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize, PartialEq)]
+pub(crate) struct ProductInfo {
+    product: Vec<Product>,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Product {
+    name: String,
+    version: String,
+    comment: Option<String>,
+}
+
+#[inline]
```

Comment:
I'm not too familiar with the rust compiler, but are inline directives useful?

I expect that the compiler can make suitable inlining decisions (especially in cases like this with small private functions), so inline directives are not helpful.